include/wil/resource.h

```diff
@@ -5241,13 +5241,19 @@ namespace wil
     {
         unique_hglobal_locked() = delete;
 
-        unique_hglobal_locked(_In_ STGMEDIUM& medium) : unique_any<void*, decltype(&::GlobalUnlock), ::GlobalUnlock>(medium.hGlobal)
+        unique_hglobal_locked(HGLOBAL global) : unique_any<void*, decltype(&::GlobalUnlock), ::GlobalUnlock>(global)
         {
             // GlobalLock returns a pointer to the associated global memory block and that's what callers care about.
-            m_globalMemory = GlobalLock(medium.hGlobal);
+            m_globalMemory = GlobalLock(global);
         }
 
-        // In the future, we could easily add additional constructor overloads such as unique_hglobal_locked(HGLOBAL) to make consumption easier.
+        unique_hglobal_locked(unique_hglobal &global) : unique_hglobal_locked(global.get())

```

Comment:
Maybe this should be `unique_hglobal const&` instead, as it's not mutating the inbound `unique_hglobal` ?

Reply:
That was a change I suggested, because it would allow to take a rvalue `unique_hglobal`, making the locked to memory invalid after the call to the constructor. A proper move constructor could be added that will also free the hglobal in the destructor, but that would complexify the class.

https://github.com/microsoft/wil/pull/84#discussion_r314894385

Reply:
No, don't add `const`